src/honeybadger/context.cr

```diff
@@ -0,0 +1,34 @@
+module Honeybadger
+  def self.context(**args) : Nil
+    Context.current.store(args)
+  end
+
+  class Context
+    def initialize
+      @data = Hash(String, String).new
+    end
+
+    delegate to_s, to: @data
+    forward_missing_to @data
+
+    def self.current : self
+      Fiber.current.honeybadger_context
+    end
+
+    private def set(key, value)
+      self.[key.to_s] = value.to_s
```

Comment:
Was the `.` after `self` a typo? I didn’t even realize you could do this.

```suggestion
      self[key.to_s] = value.to_s
```

Reply:
It's not a typo but I suppose it is superfluous regardless. Good looking out, that's a tiny detail to notice! Crystal (and ruby) operators are just methods and can be called in strange ways. This is valid in both languages:

```ruby
puts 3.+(1./(4))
```

Reply:
Yeah, I didn’t realize you could just stick the dot in there. 😄 I thought you had to call it similarly to the arithmetic operator methods as you pointed out:

```crystal
self.[]=(key.to_s, value.to_s)
```